### PR TITLE
Add operation environment variable

### PR DESCRIPTION
**Pull request by pawamoy**
First pass at #240.

### CODE REVIEW

File: copier/main.py [hunk 1]
```diff
@@ -290,6 +301,9 @@ def update_diff(conf: ConfigData) -> None:
             )
         }
     )
+    # At this point we'll run tasks in the current project:
+    # copy_local() needs to pass "update" as operation
+    os.environ["OPERATION"] = "update"
```
Comment (Yajo) [inline copier/main.py:306]: This is too much, you're altering the environment variables for all the process.

You better do:

```python
with local.env(OPERATION='update'):
   ...
```

Docs: https://plumbum.readthedocs.io/en/latest/local_machine.html#environment

**Review by Yajo — COMMENTED** [review]
It looks like a good first approach.

However I was hoping for a deeper refactoring of how tasks work. My idea was to run tasks always: pre-copy, post-copy, pre-update, post-update. The updatediff case is a special one that I'd like to review deeper, because tasks need to run there in the temporary subproject generated to obtain the diff.

Also I'm wondering if having a complex `_tasks` configuration is needed, given any script can just operate based on env variables.

What are your thoughts?

File: copier/main.py (hunk 1 above)
Comment (pawamoy) [reply copier/main.py:306]: We could also consider passing an `operation` argument to `copy_local`, defaulting to `copy`, as it's the only place we use `os.environ.get("OPERATION")`.

**Comment by pawamoy** [comment]
I've open #288 after thinking more about this.

**Comment by Yajo** [comment]
Thanks, I'll keep it in mind when fixing #240.

**Pull request closed without merging**